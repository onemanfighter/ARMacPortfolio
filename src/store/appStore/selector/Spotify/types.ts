export type SpotifyPlaylistData = {
  name: string;
  url: string;
};
