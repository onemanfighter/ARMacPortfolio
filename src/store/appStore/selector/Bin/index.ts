export { binSelector } from './Bin.selector';
