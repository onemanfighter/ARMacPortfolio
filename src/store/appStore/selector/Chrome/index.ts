export { chromeSelector } from './Chrome.selector';
