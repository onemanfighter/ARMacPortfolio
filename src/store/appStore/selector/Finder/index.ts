export { finderSelector } from './Finder.selector';
