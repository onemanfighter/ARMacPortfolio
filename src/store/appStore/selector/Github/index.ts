export { githubSelector } from './Github.selector';
