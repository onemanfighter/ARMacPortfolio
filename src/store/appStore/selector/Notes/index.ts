export { notesSelector } from './Notes.selector';
