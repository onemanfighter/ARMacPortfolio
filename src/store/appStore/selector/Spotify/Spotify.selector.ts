import { AppStoreState } from '../../appStore';

const spotifySelector = (state: AppStoreState) => ({});

export { spotifySelector };
