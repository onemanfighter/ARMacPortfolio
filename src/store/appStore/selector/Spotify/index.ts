export { spotifySelector } from './Spotify.selector';
