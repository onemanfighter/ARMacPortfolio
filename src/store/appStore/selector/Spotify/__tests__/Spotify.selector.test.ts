import { act, renderHook } from '@testing-library/react-hooks';
import { appStore } from '../../../appStore';
import { spotifySelector } from '../Spotify.selector';
import { IndianArtist } from '@appStore';

describe('Spotify selector', () => {
  it('should return default spotify state', () => {
    const { result } = renderHook(() => appStore(spotifySelector));

    expect(result.current.state).toMatchSnapshot();
  });

  it('should return current playlist data', () => {
    const { result } = renderHook(() => appStore(spotifySelector));

    expect(result.current.getCurrentPlaylist()).toMatchSnapshot();
  });

  it('should return open playlist', () => {
    const { result } = renderHook(() => appStore(spotifySelector));

    act(() => {
      result.current.openPlaylist(IndianArtist.ARIJIT_SINGH);
    });

    expect(result.current.getCurrentPlaylist).toMatchSnapshot();
  });

  it('should return close playlist', () => {
    const { result } = renderHook(() => appStore(spotifySelector));

    act(() => {
      result.current.closePlaylist();
    });

    expect(result.current.state).toMatchSnapshot();
  });

  it('should return undefined playlist data', () => {
    const { result } = renderHook(() => appStore(spotifySelector));

    act(() => {
      result.current.openPlaylist('' as IndianArtist);
    });

    expect(result.current.getCurrentPlaylist()).toBeUndefined();
  });
});
