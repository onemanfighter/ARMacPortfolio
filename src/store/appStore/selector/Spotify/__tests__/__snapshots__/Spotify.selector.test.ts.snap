// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spotify selector should return close playlist 1`] = `"main"`;

exports[`Spotify selector should return current playlist data 1`] = `
{
  "name": "A.R. Rahman",
  "url": "https://open.spotify.com/embed/artist/1mYsTxnqsietFxj1OgoGbG?utm_source=generator",
}
`;

exports[`Spotify selector should return default spotify state 1`] = `"playing"`;

exports[`Spotify selector should return open playlist 1`] = `[Function]`;
